independent-projects/vertx-java-io/src/main/java/io/quarkus/vertx/java/io/VertxBufferImpl.java

```diff
@@ -1,4 +1,4 @@
-package io.quarkus.vertx.core.runtime;
+package io.quarkus.vertx.java.io;
```

Comment:
I know that the content of `io.quarkus.vertx.core.runtime` is in theory not part of the public API but this looks like an unnecessary breaking change, or? Could we just extend a shared implementation instead? So that the clients do not need to change anything?

Maybe I'm complicating it too much...

Reply:
I'd let @geoand comment on that. From my outsider's PoV `VertxBufferImpl` seems to be a relatively simple and low-maintenance class, so we could also keep having two copies.

Reply:
It's not about having two copies but about making the `io.quarkus.vertx.VertxBufferImpl` extend the new impl so that users of the original `VertxBufferImpl` are not broken...

Reply:
I understand your concern with the public API and the proposed solution. I am just not 100% sure the gain of reusing the shared VertxBufferImpl by adding a dependency and letting the local VertxBufferImpl extend the shared one for the sake of backwards compatibility is worth the added complexity. I can do it like that if there are more voices in favor of that?

Reply:
> ...is worth the added complexity

Well, the complexity here is just about keeping one more class. The dependency will be there anyway. But I don't insist it's just a proposal ;-).

Reply:
I don't think we need to be too concerned about this breakage.

Reply:
Thanks for the feedback, @mkouba and @geoand. Let me then leave this part as it is in the current state of the PR: +dependency -BufferImpl copy.